### PR TITLE
ascanrulesBeta: add example alerts to Source Code Disclosure - Git

### PR DESCRIPTION
**Pull request by GRVial**
- Added example alerts for [Source Code Disclosure - Git](https://www.zaproxy.org/docs/alerts/41/)
- Added unit test for example alerts
- Added note to CHANGELOG

Issue Related: [zaproxy/zaproxy#6119](https://github.com/zaproxy/zaproxy/issues/6119)

### CODE REVIEW

**Comment by kingthorin** [comment]
To address the DCO requirement you'll need to sign-off the commit(s):

- https://github.com/zaproxy/zaproxy/blob/main/CONTRIBUTING.md#developer-certificate-of-origin
- https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff

**Comment by GRVial** [comment]
> To address the DCO requirement you'll need to sign-off the commit(s):
> 
>     * https://github.com/zaproxy/zaproxy/blob/main/CONTRIBUTING.md#developer-certificate-of-origin
> 
>     * https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff

**Comment by GRVial** [comment]
> To address the DCO requirement you'll need to sign-off the commit(s):
> 
>     * https://github.com/zaproxy/zaproxy/blob/main/CONTRIBUTING.md#developer-certificate-of-origin
> 
>     * https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff

Was looking into that, i belive i got it signed-off now.

**Review by kingthorin — APPROVED** [review]
Thanks

**Comment by GRVial** [comment]
I fixed the format violations during checks, but i'm getting NullPointerException on `.setUri(getBaseMsg().getRequestHeader().getURI().toString())` at SourceCodeDisclosureGitScanRule.java after running `shouldReturnExpectedExampleAlert()` unit test. Not really sure on what to look for to fix this.

**Comment by kingthorin** [comment]
There are two options I can see:
1. Pass the URL in so that you can control it for examples.
2. Check the history to see why the URL is being explicitly set. (If it’s just “always been that way” that set call can likely be removed as it’s unnecessary.)

**Comment by GRVial** [comment]
> There are two options I can see:
> 
>     1. Pass the URL in so that you can control it for examples.
> 
>     2. Check the history to see why the URL is being explicitly set. (If it’s just “always been that way” that set call can likely be removed as it’s unnecessary.)

I did what you suggested on `1` and the test now passes. Hopefully all checks will pass now.

File: addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/SourceCodeDisclosureGitScanRule.java [hunk 1]
```diff
@@ -425,4 +426,31 @@ private boolean findSourceCodeGit(HttpMessage originalMessage) throws Exception
...
+    private AlertBuilder createAlert(
+            byte[] disclosedData,
+            String filename,
+            String gitURIs,
+            HttpMessage originalMessage,
+            String uri) {
+        return newAlert()
+                .setConfidence(Alert.CONFIDENCE_MEDIUM)
+                .setUri(uri)
+                .setOtherInfo(new String(disclosedData))
```
Comment (thc202) [inline addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/SourceCodeDisclosureGitScanRule.java:439]: Make the parameter a `String` instead.

File: addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/SourceCodeDisclosureGitScanRule.java [hunk 2]
```diff
@@ -425,4 +426,31 @@ private boolean findSourceCodeGit(HttpMessage originalMessage) throws Exception
...
+                .setEvidence(getEvidence(filename, gitURIs))
+                .setMessage(originalMessage)
+                .setRisk(getRisk());
```
Comment (thc202) [inline addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/SourceCodeDisclosureGitScanRule.java:442]: Why was this added? That's the default.

File: addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/SourceCodeDisclosureGitScanRule.java [hunk 3]
```diff
@@ -425,4 +426,31 @@ private boolean findSourceCodeGit(HttpMessage originalMessage) throws Exception
...
+    }
+
+    @Override
+    public List<Alert> getExampleAlerts() {
+        return List.of(
+                createAlert(
+                                "disclosedData example".getBytes(),
```
Comment (thc202) [inline addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/SourceCodeDisclosureGitScanRule.java:449]: `"Example disclosed data."`

File: addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/SourceCodeDisclosureGitScanRule.java [hunk 4]
```diff
@@ -425,4 +426,31 @@ private boolean findSourceCodeGit(HttpMessage originalMessage) throws Exception
...
+                                "exampleFile",
```
Comment (thc202) [inline addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/SourceCodeDisclosureGitScanRule.java:450]: If we are passing a value for the file name we should pass for the URI too (but like other info, evidence is not shown in docs though).

Comment (GRVial) [reply addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/SourceCodeDisclosureGitScanRule.java:450]: Apparently i forgot to put it back in before making the commit. Will add it back (URI being empty seems to be the cause for the tests failling).

Comment (thc202) [reply addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/SourceCodeDisclosureGitScanRule.java:450]: I don't think that's the cause of the test failure, I will take a better look shortly to know what's missing.

Comment (GRVial) [reply addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/SourceCodeDisclosureGitScanRule.java:449]: I'm not super sure either, but changing it made the tests work locally. I also made the other changes requested. If you find anything let me know, and thanks for the quick replies.

File: addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/SourceCodeDisclosureGitScanRule.java [hunk 5]
```diff
@@ -400,7 +400,7 @@ private boolean findSourceCodeGit(HttpMessage originalMessage) throws Exception
                     // file on which the attack was launched.
                     // it's the least worst way of doing it, IMHO.
                     createAlert(
-                                    disclosedData,
+                                    disclosedData.toString(),
```
[the remark this answers is not in the document]
Comment (kingthorin) [reply addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/SourceCodeDisclosureGitScanRule.java:403]: I thought you suggested changing away from that earlier?

[the remark this answers is not in the document]
Comment (kingthorin) [reply addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/SourceCodeDisclosureGitScanRule.java:403]: Maybe it was a different place

[the remark this answers is not in the document]
Comment (thc202) [reply addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/SourceCodeDisclosureGitScanRule.java:403]: If you mean this https://github.com/zaproxy/zap-extensions/pull/4567#discussion_r1179972968 I suggested changing the type of the parameter, not how the array should be converted to `String`.

[the remark this answers is not in the document]
Comment (kingthorin) [reply addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/SourceCodeDisclosureGitScanRule.java:403]: Oh okay, thanks

File: addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/SourceCodeDisclosureGitScanRule.java (hunk 4 above)
Comment (GRVial) [reply addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/SourceCodeDisclosureGitScanRule.java:449]: Oh i see.

File: addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/SourceCodeDisclosureGitScanRule.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (GRVial) [reply addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/SourceCodeDisclosureGitScanRule.java]: Allright, changed it to `new String(disclosedData)` .

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you! How would you like to be credited (e.g. name, handle)?
https://www.zaproxy.org/docs/desktop/credits/#zap-extended-team

**Comment by GRVial** [comment]
> Thank you! How would you like to be credited (e.g. name, handle)? https://www.zaproxy.org/docs/desktop/credits/#zap-extended-team

Only the github name is fine. I hope to contribute more soon.
